Code/Framework/AzToolsFramework/AzToolsFramework/Archive/ArchiveComponent.cpp

```diff
@@ -12,118 +12,97 @@
+                        success = (fileSize == bytesRead);
+                    }
                 }
+
+                fileIO->Close(fileHandle);
             }
+
+            return success;
         }
 
-    private:
-        void EchoBuffer(const AZStd::string& buffer)
+        // Reads a text file that contains a list of file paths.
+        // Tokenize the file by lines.
+        // Calls the lineVisitor function for each line of the file.
+        void ProcessFileList(const AZStd::string& filePath, AZStd::function<void(AZStd::string_view line)> lineVisitor)
         {
-            size_t startIndex = 0;
-            size_t endIndex = 0;
-            const size_t bufferSize = buffer.size();
-            for (size_t i = 0; i < bufferSize; ++i)
+            AZStd::vector<char> fileBuffer;
+            if (ReadFile(filePath, AZ::IO::OpenMode::ModeText | AZ::IO::OpenMode::ModeRead, fileBuffer))
             {
-                if (buffer[i] == '\n' || buffer[i] == '\0')
-                {
-                    endIndex = i;
-                    bool isEmptyMessage = (endIndex - startIndex == 1) && (buffer[startIndex] == '\r');
-                    if (!isEmptyMessage)
-                    {
-                        AZ_Printf(s_traceName, "%s", buffer.substr(startIndex, endIndex - startIndex).c_str());
-                    }
-                    startIndex = endIndex + 1;
-                }
+                AZ::StringFunc::TokenizeVisitor(AZStd::string_view{ fileBuffer.data(), fileBuffer.size() }, lineVisitor, "\n");
             }
         }
 
-        AzFramework::ProcessCommunicator* m_communicator = nullptr;
-    };
+    } // namespace ArchiveUtils
 
     void ArchiveComponent::Activate()
     {
-        m_zipExePath = Platform::GetZipExePath();
-        m_unzipExePath = Platform::GetUnzipExePath();
+        m_localFileIO = AZ::IO::FileIOBase::GetDirectInstance();
+        if (m_localFileIO == nullptr)
+        {
+            AZ_Error(s_traceName, false, "Failed to create a LocalFileIO instance!");
+        }
 
-        ArchiveCommands::Bus::Handler::BusConnect();
+        m_archive = AZ::Interface<AZ::IO::IArchive>::Get();
+        if (m_archive == nullptr)
+        {
+            AZ_Error(s_traceName, false, "Failed to get IArchive interface!");
+        }
+
+        ArchiveCommandsBus::Handler::BusConnect();
     }
 
     void ArchiveComponent::Deactivate()
     {
-        ArchiveCommands::Bus::Handler::BusDisconnect();
+        ArchiveCommandsBus::Handler::BusDisconnect();
+
+        m_localFileIO = nullptr;
+        m_archive = nullptr;
 
-        AZStd::unique_lock<AZStd::mutex> lock(m_threadControlMutex);
-        for (auto pair : m_threadInfoMap)
+        for (AZStd::thread& t : m_threads)
         {
-            ThreadInfo& info = pair.second;
-            info.shouldStop = true;
-            m_cv.wait(lock, [&info]() {
-                return info.threads.size() == 0;
-            });
+            t.join();
         }
```

Comment:
The vector can be safely cleared now, after all the threads have been joined.

So adding a call to `m_threads = {}` can be added.

Reply:
Will clear m_threads.